### DISCUSSION TITLE
Some questions on configuring capsules, as well as how they are (or are not) weighted

### DISCUSSION THREAD

**Discussion by jeffbax** [sidekiq/sidekiq#6449] [Q&A] [answered]
Hello,

Just trying to understand a bit more about the capsule configuration/initialization. Namely in regard to max worker counts and prioritization between capsules. I've tried [parsing through the docs](https://github.com/sidekiq/sidekiq/wiki/Advanced-Options#capsules) and some of the other discussions about Capsules, but haven't quite found anything regarding the feasibility of this kind of setup.

If I have a hypothetical max of 20 workers on my Heroku dyno, and a handful of jobs that either need to run serially or at a separate concurrency than the norm -- if I configure them in capsules do they "deduct" from some total configuration? Or are all the concurrency definitions independent?

Basically a configuration that is like this in priority and with concurrency:
```
capsule [serial_1] - 1 at a time
capsule [serial_2] - 1 at a time
capsule [rate_limited] - 5 at a time
default capsule [high, default, low] -- fine running up to the max of 20
```

With `serial_1, serial_2` being the highest priority I always want to run first if any need to, followed by `rate_limited` group and after that any excess capacity goes to the normal queues that would be in the default capsule.

So that the "normal" queues could vary from using 13-20 workers depending on what the other capsules leave for them.

Is this "max" of 20 threads possible as I'm describing? Or is it really only possible to define that last grouping as `concurrency: 13` to stay under the hypothetical max of 20, because capsules aren't aware of any kind of total and themselves can't be ranked against one another?

I know you're not a fan of the YAML setup... but I didn't see anything in the Ruby setup instructions, so I wasn't sure if [this configuration](https://github.com/sidekiq/sidekiq/pull/5547/files) handles what I'm describing or if this would be `total concurrency = 53` with no ranking of capsules themselves.

Thanks!
Jeff

**Comment by mperham** [comment, marked as answer]
There's no global budget of concurrency which Sidekiq can automatically split/share between capsules. Sidekiq is rather static in its runtime in order to be simple and predictable. Autoscaling or dynamically adjusting concurrency is not something Sidekiq itself can or will ever do.

So if your additional capsules have 7 concurrency, you'd use `concurrency: 13` to get a grand total of 20 concurrency in the process.

**Reply by jeffbax** [reply]
Thanks Mike, appreciate the clarification!
